src/webapp/pages/landing/LandingPage.tsx

```diff
@@ -1,40 +1,299 @@
-                    listAction: () => history.push("/for/John"),
-                },
-                {
-                    name: "Mary",
-                    description: "Entry point 2",
-                    listAction: () => history.push("/for/Mary"),
-                },
-            ],
+    const theme = useTheme();
+    const { compositionRoot, currentUser } = useAppContext();
+
+    const [options, setOptions] = React.useState({
+        includeHeaders: true,
+        allLanguages: false,
+        allDatasets: false,
+    });
+
+    const handleChange = React.useCallback(
+        (event: React.ChangeEvent<HTMLInputElement>) => {
+            setOptions({ ...options, [event.target.name]: event.target.checked });
         },
-    ];
+        [options]
+    );
+
+    const dataSetSelectorProps = useDatasetSelector();
+
+    const exportToExcel = React.useCallback(() => {
+        compositionRoot.dataSets.export.execute([]).run(
+            _res => {
+                // console.log(res);
+            },
+            () => {}
+        );
+    }, [compositionRoot]);
+
+    const resetView = React.useCallback(() => {}, []);
+
+    const selectedDatasets = React.useMemo(() => {
+        return options.allDatasets
+            ? dataSetSelectorProps.allItems
+            : dataSetSelectorProps.allItems.filter(i => dataSetSelectorProps.values.includes(i.id));
+    }, [dataSetSelectorProps.allItems, dataSetSelectorProps.values, options.allDatasets]);
+
+    const availableLocales = React.useMemo(() => {
+        return _c(selectedDatasets).isEmpty()
+            ? []
+            : _c(
+                  selectedDatasets.flatMap(dataSet =>
+                      dataSet.translations.flatMap(t =>
+                          t.property === "NAME" ? [t.locale.split("_")[0]] : []
+                      )
+                  )
+              )
+                  .uniq()
+                  .compact()
+                  .value()
+                  .concat("en"); // Add English because is not included in translations
+    }, [selectedDatasets]);
+
+    const languageSelectorProps = useLanguagesSelector(
+        availableLocales,
+        currentUser.preferredLocale
+    );
+
+    const _selectedLocales = React.useMemo(() => {
+        return options.allLanguages
+            ? languageSelectorProps.allItems
+            : languageSelectorProps.allItems.filter(i =>
+                  languageSelectorProps.values.includes(i.code)
+              );
+    }, [languageSelectorProps.allItems, languageSelectorProps.values, options.allLanguages]);
+
+    const loading = React.useMemo(
+        () =>
+            dataSetSelectorProps.loading === "loading" ||
+            languageSelectorProps.loading === "loading",
+        [dataSetSelectorProps.loading, languageSelectorProps.loading]
+    );
+
+    React.useEffect(() => {
+        if (options.allDatasets === false) {
+            resetView();
+        }
+    }, [options.allDatasets, resetView]);
 
     return (
```

Comment:
This component is too large. Let's reduce the LOCs with the typical top-down, bottom-up strategies: split into more components, custom hooks, abstract to functions, ...